crypto/keys/src/x25519.rs

```diff
@@ -45,22 +50,25 @@ impl AsRef<[u8]> for X25519Secret {
 
 /// The debug implementation will output the SHA-256 sum of the secret, but not
 /// the secret itself
-impl Debug for X25519Secret {
-    fn fmt(&self, f: &mut Formatter) -> FmtResult {
+impl core::fmt::Debug for X25519Secret {
+    fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
         let mut hasher = Sha256::new();
         hasher.update(self.as_ref());
         let hash_results = hasher.finalize();
 
-        let output = hex::decode(&hash_results).map_err(|_e| FmtError)?;
```

Comment:
it looks to me like this was backwards..? assuming the intent was to hex _encode_ the hash